lib/galaxy/webapps/galaxy/api/library_contents.py

```diff
@@ -230,6 +230,8 @@ def create(self, trans, library_id, payload, **kwd):
         # The rest of the security happens in the library_common controller.
         real_folder_id = trans.security.encode_id(parent.id)
 
+        payload['tag_using_filenames'] = util.string_as_bool(payload.get('tag_using_filenames', None))
```

Comment:
I don't think this is 100% right, the payload will be parsed using `json.loads()`, so if you send a boolean this should be correct.
```
In [4]: json.loads(json.dumps({'really?': True}))
Out[4]: {'really?': True}
```

That said maybe we want to still allow it ?

Reply:
It's currently allowed in `POST /api/libraries/datasets` https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/webapps/galaxy/api/library_datasets.py#L378
But I'm ok with changing bioblend to send bools instead (I just don't know how :D)

Reply:
Hmm, yeah, might be easiest to just do that. I'm not sure we'd be handling something like https://stackoverflow.com/a/35946962 correctly